File: 1-Iniciante/ex1118.py
#função que faz a validação da entrada
def validacao(n):
    if n < 0 or n > 10:
        print('nota invalida')
        return 0
    else:
        return 1

#entrada
x = 1
while x == 1:
    nota1 = float(input())
    valida = validacao(nota1)
    while valida == 0:
        nota1 = float(input())
        valida = validacao(nota1)

    nota2 = float(input())
    valida = validacao(nota2)
    while valida == 0:
        nota2 = float(input())
        valida = validacao(nota2)

    #processamento
    media = (nota1 + nota2) / 2

    #saída
    print(f'media = {media:.2f}')

    while True: 
        print('novo calculo (1-sim 2-nao)')
        x = int(input())
        if x > 0 and x < 3:
            break

File: 1-Iniciante/Ex1118.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Ex1118{ // para ser aceito no Beecrowd mude o nome da classe para Main

    public static void main(String[] args) throws IOException{
        // variaveis & objetos
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

        // entrada
        int resp;
        do {
            double nota1 = validaNota(bf); // validacao
            double nota2 = validaNota(bf);

            // processamento
            double media = (nota1 + nota2) / 2;

            // saida
            System.out.printf("media = %.2f\n", media);

            resp = 0;
            while (resp < 1 || resp > 2){ // validacao resposta
                System.out.print("novo calculo (1-sim 2-nao)\n");
                resp = Integer.parseInt(bf.readLine()); // pergunta ao usuario se ele deseja continuar
            }
        } while (resp != 2);
    }

    public static double validaNota(BufferedReader bf) throws IOException{
        double nota;
        while (true){ // validacao
            nota = Double.parseDouble(bf.readLine());

            if(nota >= 0d && nota <= 10d){
                break;
            }
            System.out.print("nota invalida\n");
        }
        return nota;
    }
}
